src/helpers/fee.ts

```diff
@@ -207,3 +207,143 @@ export function getSwapAmount(
 
   return { amountOut, totalFee };
 }
+
+/**
+ * Converts basis points (bps) to a fee numerator
+ * 1 bps = 0.01% = 0.0001 in decimal
+ *
+ * @param bps - The value in basis points [1-10_000]
+ * @returns The equivalent fee numerator
+ */
+export function bpsToFeeNumerator(bps: number): number {
+  return (bps * FEE_DENOMINATOR) / BASIS_POINT_MAX;
+}
+
+/**
+ * Converts a fee numerator back to basis points (bps)
+ *
+ * @param feeNumerator - The fee numerator to convert
+ * @returns The equivalent value in basis points [1-10_000]
+ */
+export function feeNumeratorToBps(feeNumerator: number): number {
+  return (feeNumerator * BASIS_POINT_MAX) / FEE_DENOMINATOR;
+}
+
+/**
+ * Calculates the reduction factor for a Linear fee schedule
+ * Formula: fee = cliff_fee_numerator - passed_period * reduction_factor
+ *
+ * @param cliffFeeBps - The initial fee value at the cliff in basis points
+ * @param targetFeeBps - The desired final fee value after all periods in basis points
+ * @param totalPeriods - The total number of periods over which the reduction occurs
+ * @returns The estimate of reduction factor
+ */
+export function estimateLinearReductionFactor(
```

Comment:
Do we need that function?
I think user would care for base fee paramters, so we should create a function, that allow user to enter:
- max_base_fee_bps
- min_base_fee_bps
- fee scheduler mode
- number of periods
-  total duration
Then from that we calculate to base fee parameters

Reply:
pls return `BaseFee`

Reply:
updated